beaver/worker/tail.py

```diff
@@ -236,44 +242,87 @@ def _ensure_file_is_good(self, current_time):
 
     def _run_pass(self):
         """Read lines from a file and performs a callback against them"""
-        while True:
-            try:
-                data = self._file.read(4096)
-            except IOError, e:
-                if e.errno == errno.ESTALE:
-                    self.active = False
-                    return False
+        events = []
+        line_count = 0
+        buffered_lines = 0
+        buffered_bytes = 0
+        run_start = time.time()
 
-            lines = self._buffer_extract(data)
+        try:
+            data = self._file.read(self._file_read_blocksize)
+        except IOError, e:
+            if e.errno == errno.ESTALE:
+                self.active = False
+                return False
+        lines = self._buffer_extract(data)
 
+        while lines or time.time() - run_start < self._buffered_lines_max_seconds:
```

Comment:
Does this mean that when lines is None, and time.time() - run_start < self._buffered_lines_max_seconds (e.g. we set a high buffered_lines_max_seconds),  we will exit this loop and restart the tailer?

Also, why are we going off of the run_start and not the _last_activity? It seems like the buffered_lines_max_seconds is the interval between when we flush, not from now and when we started.

Reply:
_run_pass is called in a loop. If I go off of _last_activity then I will never exit _run_pass if there is consistently new data flowing in, which is not how things are designed here.

When run_pass is executed, I want to buffer until either the max_bytes, max_lines, or max_seconds is reached, and then exit the function. So from when we started this function to now is the correct thing to compare to max_seconds here.

Reply:
Oh yeah.  good point. Will change.

Reply:
_last_activity isn't exactly what we want but we do want to set run_start = time.time() each time we flush the buffer due to exceeding the line or byte count.